app/models.py

```diff
@@ -26,3 +50,38 @@ def __repr__(self):
     def html_joke(self):
         """from (string) joke --> to (array) joke divided by enters"""
         return self.joke.split('\n')
+
+    def reactions_num(self, current_enum):
+        """number of specific reaction"""
+        reactions = self.reactions.filter(JokeReaction.reaction_type == current_enum).count()
+        return reactions
+
+    def all_reactions(self):
+        """number of all reactions for specifi joke"""
+        reactions = self.reactions.count()
+        return reactions
+
+    def order_reactions(self):
+        """order reactions"""
+        unamused_num = self.reactions_num(ReactionsType.unamused)
+        neutral_num = self.reactions_num(ReactionsType.neutral)
+        smile_num = self.reactions_num(ReactionsType.smile)
+        funny_num = self.reactions_num(ReactionsType.funny)
+        reactions_data = []
+        if unamused_num != 0:
+            reactions_data.append(("unamused", unamused_num))
+        if neutral_num != 0:
+            reactions_data.append(("neutral", neutral_num))
+        if smile_num != 0:
+            reactions_data.append(("smile", smile_num))
+        if funny_num != 0:
+            reactions_data.append(("funny", funny_num))
+        reactions_data = sorted(reactions_data, key=itemgetter(1), reverse=True)
+        return reactions_data
```

Comment:
Instead of the many ifs and temp variables just append the self.reactions_num() results to the reactions_data list, then do a list comprehension: reactions_data = [item for item in reactions_data if item[1] > 0] - this filters the 0 results, then you can sort it, it'll result in a way shorter function